### PR TITLE
Consolidate logging mechanism

### PR DESCRIPTION
**Pull request by stefanceriu**
Replaced NSLog usages with a custom MXLog implementation based on SwiftyBeaver. 
- added ObjC interop through the macros from MXLog.h and an pure ObjC wrapper in order to support variable arguments
- moved logging configuration and MXLogger redirecting to the MXLog configure method
- using MXLogDebug everywhere for now, better grouping will be done later down the line

### CODE REVIEW

File: MatrixSDK/Utils/MXLogObjcWrapper.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface MXLogObjcWrapper : NSObject
```
Comment (manuroe) [inline MatrixSDK/Utils/MXLogObjcWrapper.h:21]: Class comment and its relationship with the Swift MXLog.

**Review by manuroe — COMMENTED** [review]
Can you add an input to CHANGES.rst?
Can you add a reference to the initial issue? You can even use some keyword like `fix #issue_number` to automatically close the issue on PR merge?

File: MatrixSDK.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -17,6 +17,16 @@
...
+		181FD5D62660C791008EC084 /* MXLogObjcWrapper.m in Sources */ = {isa = PBXBuildFile; fileRef = 181FD5D32660C791008EC084 /* MXLogObjcWrapper.m */; };
+		181FD5D72660C791008EC084 /* MXLogObjcWrapper.m in Sources */ = {isa = PBXBuildFile; fileRef = 181FD5D32660C791008EC084 /* MXLogObjcWrapper.m */; };
+		189CC081265E361C00BE1FD8 /* MXLog.h in Headers */ = {isa = PBXBuildFile; fileRef = 189CC07F265E361500BE1FD8 /* MXLog.h */; settings = {ATTRIBUTES = (Public, ); }; };
+		189CC082265E361D00BE1FD8 /* MXLog.h in Headers */ = {isa = PBXBuildFile; fileRef = 189CC07F265E361500BE1FD8 /* MXLog.h */; settings = {ATTRIBUTES = (Public, ); }; };
+		189CC083265E362600BE1FD8 /* MXLog.swift in Sources */ = {isa = PBXBuildFile; fileRef = 189CC080265E361500BE1FD8 /* MXLog.swift */; };
+		189CC084265E362700BE1FD8 /* MXLog.swift in Sources */ = {isa = PBXBuildFile; fileRef = 189CC080265E361500BE1FD8 /* MXLog.swift */; };
+		189CC086265E72B900BE1FD8 /* MatrixSDK-Prefix.pch in Headers */ = {isa = PBXBuildFile; fileRef = 189CC085265E72B900BE1FD8 /* MatrixSDK-Prefix.pch */; };
```
[the remark this answers is not in the document]
Comment (stefanceriu) [reply MatrixSDK.xcodeproj/project.pbxproj:28]: Nop, just a broken reference in the Xcode project.

File: MatrixSDK/Categories/MXRestClient+Extensions.swift [hunk 3]
```diff
@@ -59,11 +59,9 @@ public extension MXRestClient {
             self.downloadKeys(forUsers: chunkedUsers, token: token) { response in
                 switch response {
                     case .success(let keysQueryResponse):
-                        NSLog("[MXRestClient+Extensions] downloadKeysByChunk: Got intermediate response. Got device keys for %@ users. Got cross-signing keys for %@ users",
-                              String(describing: keysQueryResponse.deviceKeys.userIds()?.count),
-                              String(describing: keysQueryResponse.crossSigningKeys.count))
+                        MXLog.debug("[MXRestClient+Extensions] downloadKeysByChunk: Got intermediate response. Got device keys for %@ users. Got cross-signing keys for %@ users \(String(describing: keysQueryResponse.deviceKeys.userIds()?.count)) \(String(describing: keysQueryResponse.crossSigningKeys.count))")
```
[the remark this answers is not in the document]
Comment (stefanceriu) [reply MatrixSDK/Categories/MXRestClient+Extensions.swift:62]: Yes, swifty beaver only exposes non-variadic methods. I do think string interpolation is nicer though.

File: MatrixSDK/Utils/MXLogger.m [hunk 4]
```diff
@@ -221,7 +221,7 @@ static void handleUncaughtException(NSException *exception)
                     encoding:NSStringEncodingConversionAllowLossy
                        error:nil];
 
-    NSLog(@"[MXLogger] handleUncaughtException:\n%@", description);
+    MXLogDebug(@"[MXLogger] handleUncaughtException:\n%@", description);
```
Comment (manuroe) [inline MatrixSDK/Utils/MXLogger.m:224]:
```suggestion
    MXLogError(@"[MXLogger] handleUncaughtException:\n%@", description);
```

**Review by manuroe — APPROVED** [review]

**Pull request merged**